bootloader/bootloader.go

```diff
@@ -163,6 +166,30 @@ func genericSetBootConfig(systemFile string, content []byte) (bool, error) {
 	return true, osutil.AtomicWriteFile(systemFile, content, 0644, 0)
 }
 
+func genericUpdateBootConfigFromAssets(systemFile string, assetName string) error {
+	currentBootConfigEdition, err := editionFromDiskConfigAsset(systemFile)
+	if err != nil && err != errNoEdition {
+		return err
+	}
+	if err == errNoEdition {
+		return nil
+	}
+	newBootConfig := assets.Internal(assetName)
+	if len(newBootConfig) == 0 {
+		return fmt.Errorf("no boot config asset with name %q", assetName)
+	}
+	gbs, err := configAssetFrom(newBootConfig)
+	if err != nil {
+		return err
+	}
+	if gbs.Edition() <= currentBootConfigEdition {
+		// edition of the candidate boot config is lower than or equal
+		// to one currently installed
+		return nil
+	}
+	return osutil.AtomicWriteFile(systemFile, gbs.Raw(), 0644, 0)
```

Comment:
Nitpick, for accountability I would love to see a syslog/stdout message here.

Reply:
Our logging and auditing story is not so great. I think I can do some tweaks in a follow up that indicate whether an update took place or not.